data/src/main/java/nl/tehdreamteam/se42/data/repository/HibernateRepository.java

```diff
@@ -90,41 +92,45 @@ protected void remove(T object) {
 
     /**
      * Performs a {@link EntityTransaction} using a given function. This method performs the following steps, in order:
-     * 1. Start a {@code EntityTransaction} (See {@link #beginTransaction(EntityManager)}.
+     * 1. Start a {@code EntityTransaction} (See {@link #beginTransaction()}.
      * 2. Create a {@link DAO} (See {@link #createDao(EntityManager)}.
      * 3. Execute the given {@code function}.
-     * 4. Commit and close the {@code EntityTransaction} (See {@link #commitAndCloseTransaction(EntityManager)}.
+     * 4. Commit the {@code EntityTransaction} (See {@link #commitTransaction()}).
      *
      * @param function The {@code function} to execute in a {@code EntityTransaction}.
      * @param <R>      The return type of the given {@code function}.
      * @return An object fetched from the database, conform the requested type ({@code <R>}).
      */
     protected <R> R performTransaction(Function<D, R> function) {
-        EntityManager manager = factory.createEntityManager();
-        beginTransaction(manager);
+        try {
+            beginTransaction();
+
+            D dao = createDao(manager);
+            R value = function.apply(dao);
 
-        D dao = createDao(manager);
-        R value = function.apply(dao);
+            commitTransaction();
 
-        commitAndCloseTransaction(manager);
+            return value;
+        } catch (Exception e) {
+            rollbackTransaction();
 
-        return value;
+            throw e;
+        }
     }
 
-    private void beginTransaction(EntityManager manager) {
+    private void beginTransaction() {
         manager.getTransaction().begin();
     }
 
-    private void commitAndCloseTransaction(EntityManager manager) {
-        commitTransaction(manager);
-        closeManager(manager);
+    private void commitTransaction() {
+        manager.getTransaction().commit();
     }
 
-    private void commitTransaction(EntityManager manager) {
-        manager.getTransaction().commit();
+    private void rollbackTransaction() {
+        manager.getTransaction().rollback();
     }
 
-    private void closeManager(EntityManager manager) {
+    private void closeManager() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "closeManager" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.tehmessiah.nl/coding_rules#rule_key=squid%3AUnusedPrivateMethod)